### PR TITLE
token unit test

### PR DESCRIPTION
**Pull request by the-first-elder**
**Issue(s):** 
add tests https://github.com/NethermindEth/StarknetByExample/issues/70
### Description

Please provide a brief description of the changes made in this pull request and how they address the related issue.

### Checklist

- [x] **Contract Tests:** Added tests to cover the changes

### CODE REVIEW

**Review by julio4 — CHANGES_REQUESTED** [review]
LGTM! Thank you, I requested some small changes!

File: listings/applications/erc20/src/lib.cairo [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mod token;
+pub mod token;
 
 #[cfg(test)]
 mod tests;
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/lib.cairo]: I have moved this to the contracts module as requested

File: listings/applications/erc20/src/tests.cairo [hunk 2]
```diff
@@ -1,2 +1,359 @@
...
+            .unwrap_syscall();
+
+        (IERC20Dispatcher { contract_address }, contract_address)
+    }
+
+    fn recipient_address() -> ContractAddress {
+        let recipient: ContractAddress = contract_address_const::<'recipient'>();
+
+        recipient
+    }
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/tests.cairo]: Done

File: listings/applications/erc20/src/tests.cairo [hunk 3]
```diff
@@ -1,2 +1,359 @@
...
+
+    fn spender_address() -> ContractAddress {
+        let spender = contract_address_const::<'spender'>();
+        spender
+    }
+
+    fn zero_address() -> ContractAddress {
+        get_caller_address()
+    }
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/tests.cairo]: Done

File: listings/applications/erc20/src/tests.cairo [hunk 4]
```diff
@@ -1,2 +1,359 @@
...
+
+    #[test]
+    #[available_gas(2000000)]
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/tests.cairo]: I removed available_gas

File: listings/applications/erc20/src/tests.cairo [hunk 5]
```diff
@@ -1,2 +1,359 @@
...
+    use starknet::testing::{set_contract_address, set_account_contract_address};
+
+
+    fn deploy() -> (IERC20Dispatcher, ContractAddress) {
+        let recipient: ContractAddress = contract_address_const::<'recipient'>();
+
+        let token_name: felt252 = 'myToken';
+        let decimals: felt252 = 18;
+        let initial_supply: felt252 = 100000;
+        let symbols: felt252 = 'mtk';
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/tests.cairo]: Done

File: listings/applications/erc20/src/tests.cairo [hunk 6]
```diff
@@ -1,2 +1,359 @@
...
+        let value = 100;
+
+        let (dispatcher, contract_address) = deploy();
+        let caller = contract_address_const::<'caller'>();
+        set_contract_address(caller);
+        dispatcher.approve(spender, value);
+        set_contract_address(contract_address);
+
+        // Notice the order: the first event emitted is the first to be popped.
+        /// ANCHOR: test_events
```
[the remark this answers is not in the document]
Comment (the-first-elder) [reply listings/applications/erc20/src/tests.cairo]: done

**Comment by the-first-elder** [comment]
I have made the required changes

**Review by Mayowa002 — COMMENTED** [review]
.

File: listings/applications/erc20/src/tests.cairo
Comment (julio4) [inline listings/applications/erc20/src/tests.cairo:1]: you can safely remove this file now, and remove the `mod tests` in the lib.cairo file

**Review by julio4 — CHANGES_REQUESTED** [review]

**Review by julio4 — APPROVED** [review]

**Pull request merged**